zp-relayer/txProcessor.ts

```diff
@@ -89,3 +95,27 @@ export async function processTx(tx: TxPayload) {
   })
   return { data, commitIndex, rootAfter }
 }
+
+export async function processDirectDeposits(directDeposits: DirectDeposit[]) {
```

Comment:
What prevents simultaneous call to `processTx` and `processDirectDeposits`, thus different transaction generation for the same tree index?

Reply:
Now, all workers (there are 3 of them) use mutex to prevent race conditions, so at any point in time there is only one worker processing a transaction. But, I will remove DD worker today to properly handle transactions' priority. PoolTxWorker will handle both types of transactions: DD and normal txs.